### DISCUSSION TITLE
how pirntf the Multiranger's all tof distances ?

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1685] [General]
Hi,

how pirntf the Multiranger's all tof distances real time ?

**Comment by ataffanel** [comment]
You can log the distances at regular intervals using the log subsystem (see log examples in the lib).

If you need synchronous readings for each measurements, you can try to print the measurement in the console from the multiranger deck driver loop, though this might be too much data. Another option is to send data back in binary using an app channel data packet. This will require some custom firmware and python code but it will be by far the fastest and most efficient.

Tell me if you need more precision for any of these solution and if you need pointers to where to look in the code.

**Reply by justinleeyang** [reply]
Hi, ataffanel：

Please pointer to where in the code, Thanks

**Reply by ataffanel** [reply]
Hi,

No problem for providing you printers but I suggested a brunch of different solution and it depends greatly of your needs. Do you need synchronous or asynchronous reading? Maybe you can tell us a bit more what are your goals and we can help you figure out the best solution.

**Reply by justinleeyang** [reply]
Hi, ataffanel:

I haved get the data from the plot, Thanks!

![image](https://github.com/user-attachments/assets/ded9bba2-d46a-40b5-9eb6-78f819a56e78)
